From 10fcda985c8ba62c5192640f88adbe3db2cc2a9e Mon Sep 17 00:00:00 2001
From: Tirumalesh Chalamarla <tchalamarla@caviumnetworks.com>
Date: Tue, 9 Oct 2018 00:41:18 +0300
Subject: [PATCH 0254/1051] octeontx: fix create_domain

fix a crash with create_domain when invalid format string is given

Signed-off-by: Tirumalesh Chalamarla <tchalamarla@caviumnetworks.com>
Signed-off-by: Yury Norov <ynorov@caviumnetworks.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 .../ethernet/cavium/octeontx-83xx/octeontx_main.c  | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/drivers/net/ethernet/cavium/octeontx-83xx/octeontx_main.c b/drivers/net/ethernet/cavium/octeontx-83xx/octeontx_main.c
index 7fceef2767d0..b43b181fc789 100644
--- a/drivers/net/ethernet/cavium/octeontx-83xx/octeontx_main.c
+++ b/drivers/net/ethernet/cavium/octeontx-83xx/octeontx_main.c
@@ -143,33 +143,47 @@ static ssize_t octtx_create_domain_store(struct device *dev,
 
 		if (!strncmp(start, "ssow", sizeof("ssow") - 1)) {
 			temp = strsep(&start, ":");
+			if (!start)
+				goto error;
 			if (kstrtol(start, 10, &ssow_count))
 				goto error;
 		} else if (!strncmp(start, "fpa", sizeof("fpa") - 1)) {
 			temp = strsep(&start, ":");
+			if (!start)
+				goto error;
 			if (kstrtol(start, 10, &fpa_count))
 				goto error;
 		} else if (!strncmp(start, "sso", sizeof("sso") - 1)) {
 			temp = strsep(&start, ":");
+			if (!start)
+				goto error;
 			if (kstrtol(start, 10, &sso_count))
 				goto error;
 		} else if (!strncmp(start, "pko", sizeof("pko") - 1)) {
 			temp = strsep(&start, ":");
+			if (!start)
+				goto error;
 			if (kstrtol(start, 10, &pko_count))
 				goto error;
 		} else if (!strncmp(start, "pki", sizeof("pki") - 1)) {
 			continue;
 		} else if (!strncmp(start, "tim", sizeof("tim") - 1)) {
 			temp = strsep(&start, ":");
+			if (!start)
+				goto error;
 			if (kstrtol(start, 10, &tim_count))
 				goto error;
 		} else if (!strncmp(start, "net", sizeof("net") - 1)) {
 			temp = strsep(&start, ":");
+			if (!start)
+				goto error;
 			if (kstrtol(start, 10, &bgx_port[bgx_count]))
 				goto error;
 			bgx_count++;
 		} else if (!strncmp(start, "virt", sizeof("virt") - 1)) {
 			temp = strsep(&start, ":");
+			if (!start)
+				goto error;
 			if (kstrtol(start, 10, &lbk_port[lbk_count]))
 				goto error;
 			lbk_count++;
-- 
2.17.1

